src/redux/actions/experiments/updateExperiment.js

```diff
@@ -47,38 +52,55 @@ const updateExperiment = (
 
       const json = await response.json();
       throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
+    } else if (config.currentApiVersion === api.V2) {
+      if (_.isUndefined(experimentDiff.sampleIds)) {
```

Comment:
why is this check `isUndefined` necessary?

Reply:
I see that if `_.isUndefined` is false, `fetchAPI` won't be called, but `EXPERIMENTS_UPDATED` and `EXPERIMENTS_SAVED` actions will still be dispatched. Can you explain why is that?

Reply:
This handling wasn't well written, we discussed with iva and decided to throw an error if a dev tries to set `sampleIds`/`samplesOrder` through this action in v2 because we want the handling of this json array to be handled through specific endpoints (for reordering samples for example) or by the api directly as a side effect to whatever happens to the samples